### ISSUE TITLE
Possible Bug: EXIF Handling?

### ISSUE THREAD

**Issue by ews-grmunjal** [roboflow/rf-detr#189]
Hello,

Thank you for your awesome work! I've been toying around with it and today noticed I was getting pretty bad metrics  on one of my own benchmarks (custom data in coco format). In looking further, I found nearly all issues were EXIF related, the original images have EXIF data but I think (!) it's not accounted for in the repo? I searched and couldnt find explicit handling.

I confirmed this by applying the correct transform myself on the raw images and then using those images to train and things work well now.

**Comment by isaacrob-roboflow** [comment]
yeah we're not doing anything to handle EXIF data. can you say more about your usecase?

**Comment by ews-grmunjal** [comment]
I'm trying to train detection models using pictures collected with many different types of cameras/phones and bbox labels. The labelling software, when reading in images does EXIF handling so labels are created on the oriented picture. 

When feeding those labels and raw pictures into rfdetr for Training performance seemed very poor. Visualizing the detections on my valid set made me see the detections had some type of rotational issue.

So looked into the codebase for exif related stuff and didnt see anything. So then as a quick test, oriented all my pictures and removed their exif data and re-ran everything, it worked great. Made me think issue is PIL reading related.

**Comment by SkalskiP** [comment]
@ews-grmunjal  The reason behind this issue is that internally we're relying on [`torchvision.datasets.CocoDetection`](https://pytorch.org/vision/main/generated/torchvision.datasets.CocoDetection.html) to load COCO-formatted datasets. Specifically, the images are loaded with `Image.open(...).convert("RGB")`, a method that does not automatically apply EXIF orientation corrections, resulting in improperly oriented images when EXIF rotation data is present. Because labels are created assuming correct image orientation, discrepancies arise between the loaded images and their corresponding annotations, causing poor model performance.

@isaacrob-roboflow, a potential fix might be overriding the internal `_load_image` method of `CocoDetection` to explicitly apply EXIF corrections using something like `ImageOps.exif_transpose`. However, this would require ensuring it doesn't inadvertently break internal functionalities within `CocoDetection`, so it should be carefully evaluated and tested before adoption.

**Comment by ews-grmunjal** [comment]
@SkalskiP yes exactly, ImageOps.exif_transpose as a preprocess before rfdetr was my workaround

**Comment by isaacrob-roboflow** [comment]
Honestly this feels somewhat out of scope to me. I think we should assume incoming data is well-formatted. I worry that doing this may have weird side effects and I also worry that it'll open a precedent for doing with other rare formatting issues that other research repos also don't address. Notably, using roboflow to label would resolve this issue ;)

I'm open to having my mind changed on that though. Does ultralytics address this?

**Comment by ews-grmunjal** [comment]
Yes this is not an issue that happens with their/ultralytics repo. It works there out of the box. My incoming data is coming from make-sense ;

But happy to close issue as resolved for me with the pre-process

**Comment by SkalskiP** [comment]
@ews-grmunjal [makes sense](https://github.com/SkalskiP/make-sense)? Really? 🤣 Not to brag, but... you do know I built this tool, right? Would be nice to return to working on it someday.

@isaacrob-roboflow I agree there might be some "weird side effects," but I think we should aim to be better and more user-friendly than typical research repositories.

**Comment by isaacrob-roboflow** [comment]
@SkalskiP if our goal is to target replacing ultralytics rather than being a strong research repo, then I'm down to have that change happen, as long as it is noted somewhere that we are using that data and work is done to verify it doesn't have side effects .. did you end up working on test case stuff?

**Comment by ews-grmunjal** [comment]
@SkalskiP yes I'm aware =] it's a frictionless ux that i really like using!

Thank you both for the active discussion. I think a little unfair to imply that strong research and application are at odds with each other.

In many repos, loading happens with cv2 which by default orients images at read time but really I do not like mixing PIL with cv2

Also see related: https://github.com/pytorch/vision/issues/7977
